shiny.qmd

```diff
@@ -0,0 +1,276 @@
+# shiny(live)!
+
+The Quarto doc will contain the `shiny` app using a code chunk. The type of code chunk should be `{shinylive-r}` (as opposed to `{r}`). 
+
+## yml options
+
+The main requirement for the code chunk yml is the option `standalone: true`. 
+
+For my application, I used the options: 
+
+```yml
+#| label: fig-shiny-spline
+#| viewerHeight: 500
+#| standalone: true
+```
+
+The `label` doesn't make it a figure as it normally would but see the section below for details regarding this (and details like captions, etc.). 
+
+The standard `fig-height` (and width) options don't apply to `shiny` apps; `viewerHeight` is used instead.  
+
+You also have to add a Quarto _filter_ in your `_quarto.yml` file: 
+
+```yml
+filters:
+  - shinylive
+```
+
+There is a bit more to say about formatting the chunk for Quarto but we'll show that later. 
+
+## Package declarations
+
+In our chunk that contains the `shiny` app (i.e., `fig-shiny-spline` in this example), the first few lines should install packages that we need (besides `shiny`): 
+
+```{r}
+#| label: pkg-installs
+#| eval: false
+
+webr::install("ggplot2")
+webr::install("dplyr")
+```
+
+There is no CRAN package called `webr` so don't worry about installing that. 
```

Comment:
Akwardly, this is not true: https://cran.r-project.org/web/packages/webr/index.html

So perhaps,

```suggestion
The base installation of webR includes its own `webr` utility package, so don't worry about installing that. 
```

At the time of writing, webR's `webr` package contains routines that are only useful when running under WebAssembly -- it is likely that it will never be uploaded to CRAN. If that day ever comes, we could consider a separate package named something like `webrutils`.

In the future, I'd like shinylive to abstract this away so that users don't need to invoke `webr::` functions directly themselves anyway.

Reply:
If `webr` is automatically attached in the session, you might think about changing the name to `webr_install()` and just use that. 

It would help with potential name conflicts and, if you don't have to call by namespace, it de-emphasizes the idea that there is an extra package rolling around in there.